app/security.py

```diff
@@ -113,10 +113,10 @@ async def context_processor(request):
     }
 
 
-async def check_permission(request):
+async def check_permission(request) -> Session:
     """
     Check request permission.
-    Raise HTTPForbidden if not previously remembered.
+    Raise HTTPForbidden if not previously remembered, else return session.
     """
     session = await get_session(request)
```

Comment:
Yes the session could time out. If the Session times out they need to re-enter their UAC. All sessions with an Identity remembered will have entered a UAC. Not all sessions will have an Identity? The requirement for what happens if they do not have a valid identity is not down to me but it does raise a different error which allows us to deal with it differently.

This function could for instance call get_existing_session instead of get_session prior to checking for an Identity to show they have entered a UAC. That would result in different behaviour in the response to the user and is a requirement rather than coding question?

Reply:
Changed to follow described behaviour. It now calls get_existing_session so will display the session timeout. Only if a session is there but no identity will it now raise the forbidden error.
Further unit tests added for this change in behaviour.